eng/AvoidRestoreCycleOnSelfReference.targets

```diff
@@ -0,0 +1,14 @@
+<Project>
+  <!-- Set PackageId to a temporary value to avoid NuGet restore cycle error NU1108: https://github.com/NuGet/Home/issues/6754 -->
```

Comment:
Where were we hitting this?

Reply:
in DiagnosticSource which is an implicit package of NetStandard.Library. Discussed this with @ericstj and based on the NuGet issue this seems to be the cleanest approach for now. You can see that I set that new property in both the ref and src DiagnosticSource projects.